xcdat/spatial.py

```diff
@@ -742,3 +742,187 @@ def _averager(
+            and have the same dimensional sizes as the data variable, by default
+            "generate".
+        keep_weights : bool, optional
+            If calculating averages using weights, keep the weights in the
+            final dataset output, by default False.
+        lat_bounds : Optional[RegionAxisBounds], optional
+            A tuple of floats/ints for the regional latitude lower and upper
+            boundaries. This arg is used when calculating axis weights, but is
+            ignored if ``weights`` are supplied. The lower bound cannot be
+            larger than the upper bound, by default None.
+        lon_bounds : Optional[RegionAxisBounds], optional
+            A tuple of floats/ints for the regional longitude lower and upper
+            boundaries. This arg is used when calculating axis weights, but is
+            ignored if ``weights`` are supplied. The lower bound can be larger
+            than the upper bound (e.g., across the prime meridian, dateline), by
+            default None.
+
+        Returns
+        -------
+        xr.DataArray
+            Dataset with the spatially averaged variable.
+
+        Examples
+        --------
+
+        Check the 'axis' attribute is set on the required coordinates:
+
+        >>> da.lat.attrs["axis"]
+        >>> Y
+        >>>
+        >>> da.lon.attrs["axis"]
+        >>> X
+
+        Set the 'axis' attribute for the required coordinates if it isn't:
+
+        >>> da.lat.attrs["axis"] = "Y"
+        >>> da.lon.attrs["axis"] = "X"
+
+        Call spatial averaging method:
+
+        >>> da.spatial.average(...)
+
+        Get global average time series:
+
+        >>> ts_global = da.spatial.average(axis=["X", "Y"])["tas"]
+
+        Get time series in Nino 3.4 domain:
+
+        >>> ts_n34 = da.spatial.average(axis=["X", "Y"],
+        >>>     lat_bounds=(-5, 5),
+        >>>     lon_bounds=(-170, -120))["ts"]
+
+        Get zonal mean time series:
+
+        >>> ts_zonal = da.spatial.average(axis=["X"])["tas"]
+
+        Using custom weights for averaging:
+
+        >>> # The shape of the weights must align with the data var.
+        >>> weights = xr.DataArray(
+        >>>     data=np.ones((4, 4)),
+        >>>     coords={"lat": self.ds.lat, "lon": self.ds.lon},
+        >>>     dims=["lat", "lon"],
+        >>> )
+        >>>
+        >>> ts_global = ds.spatial.average("tas", axis=["X", "Y"],
+        >>>     weights=weights)["tas"]
+        """
+        # convert dataarray to a dataset
+        da = self._dataarray.copy()
+        ds = boundedDataArray_to_dataset(da)
+        # get data_var key
+        data_var = da.name
+        # pass on call to spatial averager
+        ds_sa = ds.spatial.average(
+            data_var,
+            axis=axis,
+            weights=weights,
+            keep_weights=keep_weights,
+            lat_bounds=lat_bounds,
+            lon_bounds=lon_bounds,
```

Comment:
This is the template: take your "bounded dataarray", convert it back to a dataset, and then call the existing xcdat API (passing along all the underlying arguments). 

A shortfall, beyond replicating most (but not all) documentation, is that if we change the dataset API, we need to make sure to update it here, too. Maybe there is a fancy way to pass in `**kwargs` that can do this automatically (probably not)?